lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceManagerStatus.java

```diff
@@ -147,7 +149,7 @@ private synchronized void onRMDone(final RuntimeStatusEvent runtimeStatusEvent)
     this.driverStatusManager.onComplete();
   }
 
-  private synchronized void onRMRunning(final RuntimeStatusEvent runtimeStatusEvent) {
+  private void onRMRunning(final RuntimeStatusEvent runtimeStatusEvent) {
```

Comment:
If `onRMRunning()` and `this.isIdle()` use the same lock, how does this change anything? The `assert` should not cause the deadlock, right?